### DISCUSSION TITLE
Either the index store is outdated

### DISCUSSION THREAD

**Discussion by trungld123** [peripheryapp/periphery#407] [Q&A]
"The index store at '/Users/Abc/Library/Caches/com.github.peripheryapp/DerivedData-a4afb026-5ddbe9bd/Index/DataStore' does not contain data for the following targets: "abcApp". Either the index store is outdated, or you have requested to scan targets that have not been built."
i dont know error when run "periphery scan --setup" at terminal command line

**Comment by ileitch** [comment]
Is the `abcApp` target a build dependency of the schemes you selected?

**Reply by jeff-iphone** [reply]
I have had same issue, it seems to be Xcode 14 related, I also get [index:swift] Source file not indexed:, but Xcode compiles and build project Ok,

**Reply by Athleteg** [reply]
> "The index store at '/Users/Abc/Library/Caches/com.github.peripheryapp/DerivedData-a4afb026-5ddbe9bd/Index/DataStore' does not contain data for the following targets: "abcApp". Either the index store is outdated, or you have requested to scan targets that have not been built."
> i dont know error when run "periphery scan --setup" at terminal command line

First of all, @ileitch  thank you. Your assumption helped to understand the problem and then to solve it.

I hope it can be useful for everyone who faced with this issue.
Well, @trungld123 I had the same issue.

**Possible steps for solving issue:**
Open project in Xcode -> tap on the project -> Choose needed target -> Tap on the `Build Phases` tab -> Tap on the `Target Dependencies` pop-up.

_Then add all these target names into your ".periphery.yml" file (below the `targets:` field)._

**Comment by jadhav-sam** [comment]
Hi @ileitch, I am also getting the same error. I did not understand your question. We have MyApp as target and SomethingElse as scheme. Also we use .xcworkspace

> Is the abcApp target a build dependency of the schemes you selected?


@ArtemUstinov, Thanks for providing the steps. What I need to add under `Target Dependencies`? I did not have `.periphery.yml`. Should I create one?

**Reply by Athleteg** [reply]
> Should I create one?

I'm not sure exactly but try to perform step by step this [guide](https://github.com/peripheryapp/periphery#the-scan-command). 
In order to see the hidden files (`.periphery` is one of that) tap such combination: `Command + Shift + .`

> What I need to add under Target Dependencies?

Target dependencies pop-up is just clarifying for you which targets are used.
Add needed for you targets to `.periphery` file under the `targets` field.
